packages/clerk-js/src/ui/components/SignIn/SignInSocialButtons.tsx

```diff
@@ -18,24 +34,57 @@ export const SignInSocialButtons = React.memo((props: SocialButtonsProps) => {
   const signIn = useCoreSignIn();
   const redirectUrl = buildSSOCallbackURL(ctx, displayConfig.signInUrl);
   const redirectUrlComplete = ctx.afterSignInUrl || displayConfig.afterSignInUrl;
+  const [captchaToken, setCaptchaToken] = useState('');
+
+  const isCaptchaEnabled = true; // TODO: replace this when environment field is ready
+  const captchaRef = useRef(null);
+  React.useLayoutEffect(() => {
+    if (isCaptchaEnabled) {
+      void loadCaptcha().then(t => {
+        t.execute(captchaRef.current, {
+          sitekey: '1x00000000000000000000AA', // test sitekey
+          // sitekey: '0x4AAAAAAAFHxLeVBtmN8VhF', // staging sitekey
+          callback: function (token: string) {
+            setCaptchaToken(token);
+            console.log(`Challenge Success ${token}`);
+          },
+        });
+      });
+    }
+  }, []);
 
   return (
-    <SocialButtons
-      {...props}
-      oauthCallback={strategy => {
-        return signIn
-          .authenticateWithRedirect({ strategy, redirectUrl, redirectUrlComplete })
-          .catch(err => handleError(err, [], card.setError));
-      }}
-      web3Callback={() => {
-        return clerk
-          .authenticateWithMetamask({
-            customNavigate: navigate,
-            redirectUrl: redirectUrlComplete,
-            signUpContinueUrl: ctx.signUpContinueUrl,
-          })
-          .catch(err => handleError(err, [], card.setError));
-      }}
-    />
+    <>
+      <SocialButtons
+        {...props}
+        oauthCallback={strategy => {
+          return signIn
+            .authenticateWithRedirect({
+              strategy,
+              redirectUrl: withQueryParamIfCaptchaIsEnabled(redirectUrl, isCaptchaEnabled, {
```

Comment:
I'd expect that we calculate the captchToken after the redirection

Reply:
@panteliselef yes, you are right. The main issue here is that a user might sign-in with Google before the token is generated. 

But the problem is that after the redirection you are not on the signUp/In page anymore. The `signUp.create` is triggered in another route(`/sso-callback`) and then redirect to whatever path the user has provided.

Another possible solution would be to trigger the token generation when a user clicks on the social button

Reply:
@raptisj  Somewhere below I suggest that the token is generated inside `.create`, wouldn't that solve the issue ?

Reply:
@panteliselef if you are referring inside the `create` in `packages/clerk-js/src/core/resources/SignUp.ts` no because we have to render the widget somewhere(even though it is invisible, still)